lib/gateway/Shard.js

```diff
@@ -2243,7 +2243,9 @@ class Shard extends EventEmitter {
                         m.member.id = m.member.user.id;
                         const guild = this.client.guilds.get(packet.d.guild_id);
                         if(guild) {
-                            guild.members.update(m.presence, guild);
+                            if(m.presence) {
+                                guild.members.update(m.presence, guild);
+                            }
                             guild.members.update(m.member);
```

Comment:
This will throw a "Missing client in constructor" in user structure too
`guild.members.update(m.member, guild);`

According to other places where the member is update, this should be:
`guild.members.update(m.member, guild);`

Reply:
Unrelated to this PR

Reply:
Do I need to open a pr for this? ><

Reply:
Making a new PR should be fine